File: Creational/Prototype/Java/Example/Shape.java
/**
 * Shape represents a geometric shape prototype
 * Demonstrates cloning with different shape types
 */
public abstract class Shape implements Prototype {
    protected String color;
    protected int x, y;
    
    public Shape() {}
    
    public Shape(Shape other) {
        if (other != null) {
            this.color = other.color;
            this.x = other.x;
            this.y = other.y;
        }
    }
    
    public abstract Prototype clone();
    public abstract String getInfo();
    public abstract double getArea();
    
    public void setColor(String color) {
        this.color = color;
    }
    
    public void setPosition(int x, int y) {
        this.x = x;
        this.y = y;
    }
    
    public String getColor() {
        return color;
    }
}

File: Creational/Prototype/Php/Example/Shape.php
<?php

/**
 * Shape represents a geometric shape prototype
 * Demonstrates cloning with different shape types
 */
abstract class Shape implements Prototype
{
    protected string $color;
    protected int $x;
    protected int $y;
    
    public function __construct()
    {
        $this->color = '';
        $this->x = 0;
        $this->y = 0;
    }
    
    public abstract function clone(): Prototype;
    public abstract function getInfo(): string;
    public abstract function getArea(): float;
    
    public function setColor(string $color): void
    {
        $this->color = $color;
    }
    
    public function setPosition(int $x, int $y): void
    {
        $this->x = $x;
        $this->y = $y;
    }
    
    public function getColor(): string
    {
        return $this->color;
    }
}